### PR TITLE
More cherry-picking

### PR DESCRIPTION
**Pull request by kornelski**

### CODE REVIEW

File: src/codec/decoder/audio.rs [hunk 1]
```diff
@@ -10,6 +10,11 @@ use crate::{
 pub struct Audio(pub Opened);
 
 impl Audio {
+	#[deprecated(
```
Comment (meh) [inline src/codec/decoder/audio.rs:13]: I'm not super convinced about this and the addition `send_packet` and friends, I feel like `decode` should just adapt to the non-deprecated variants.

What's your thought on this?

**Review by meh — CHANGES_REQUESTED** [review]

File: src/codec/decoder/audio.rs (hunk 1 above)
Comment (kornelski) [reply src/codec/decoder/audio.rs]: Yeah, makes sense.

**Comment by kornelski** [comment]
I've removed the deprecation

**Review by meh — APPROVED** [review]

**Comment by meh** [comment]
Dammit, I fudged that up, I'm just gonna do this manually.

**Pull request closed without merging**